src/main/java/org/example/Samsung.java

```diff
@@ -0,0 +1,138 @@
+package org.example;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Samsung implements Phone {
+
+    List<Contact> contactList = new ArrayList<>();
+    List<String> callHistoryList = new ArrayList<>();
+    Map<String, ArrayList<String>> messageHistoryMap = new HashMap<>();
+    Map<String, ArrayList<String>> callHistoryMap = new HashMap<>();
+    final int batteryLife;
+    int currentBatteryLife;
+    String color;
+    String material;
+    final String IMEI;
+
+    Samsung(int batteryLife, int currentBatteryLife, String color, String material, String IMEI) {
+        this.batteryLife = batteryLife;
+        this.currentBatteryLife = currentBatteryLife;
+        this.color = color;
+        this.material = material;
+        this.IMEI = IMEI;
+    }
+
+    public void addContact(String contactNumber, String phoneNumber, String firstName, String lastName) {
+        this.contactList.add(new Contact(contactNumber, phoneNumber, firstName, lastName));
+        this.messageHistoryMap.put(phoneNumber, new ArrayList<String>());
+        this.callHistoryMap.put(phoneNumber, new ArrayList<String>());
+    }
+
+    public void sendMessage(String phoneNumber, String messageContent) {
+        if (messageContent.toCharArray().length >= 500) {
+            System.out.println("Message is too long");
+            return;
+        }
+        System.out.println("Sent message to:" + phoneNumber + ":" + messageContent);
+        messageHistoryMap.get(phoneNumber).add(messageContent);
+        currentBatteryLife -= 1;
+    }
+
+    public void getMessageListForContact(String phoneNumber) {
+        if (!messageHistoryMap.containsKey(phoneNumber)) {
+            System.out.println("Contact not found");
+            return;
+        }
+        for (int i = 0; i < messageHistoryMap.get(phoneNumber).size(); ++i) {
+            System.out.println(messageHistoryMap.get(phoneNumber).get(i));
+        }
+    }
+
+    public void call(String phoneNumber) {
+        if (!callHistoryMap.containsKey(phoneNumber)) {
+            System.out.println("Contact not found");
+            return;
+        }
+        System.out.println("Calling:" + phoneNumber);
+        callHistoryMap.get(phoneNumber).add(phoneNumber);
```

Comment:
fix the map to store the date and time of the each call, a map of {phoneNumber: {date1, date2, etc}}